network/manager.go

```diff
@@ -216,7 +216,7 @@ func (mgr *Manager) StartConnector() {
 // StartListener will start a listener on a given port.
 func (mgr *Manager) StartListener(stop <-chan struct{}) {
 	mgr.wg.Add(1)
-	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr, stop)
+	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr, func(addr *net.TCPAddr) (Listener, error) { return createTCPListener(addr) }, stop)
```

Comment:
Could you just move this into its own line? It's just a mess like this. You can do
`createListener := func(addr *net.TCPAddr) (Listener, error) { ...}`
and then pass `createListener` as parameter.

Reply:
Okay, I misunderstood what you meant before. It is called extract variable.